backend/GymBeam/QueryHandlers/ActivityQueryHandler.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Domain;
+using GymBeam.Queries;
+using GymBeam.Responses;
+using MediatR;
+
+namespace GymBeam.QueryHandlers
+{
+    public class ActivityQueryHandler : IRequestHandler<GetAllActivitiesQuery, IEnumerable<ActivityResponse>>
+    {
+        private readonly IRepository _repository;
+
+        public ActivityQueryHandler(IRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public Task<IEnumerable<ActivityResponse>> Handle(GetAllActivitiesQuery request, CancellationToken cancellationToken)
+        {
+            var activities = _repository.GetActivities();
+
+            if (activities == null || !activities.Any())
+            {
+                return Task.FromResult(Enumerable.Empty<ActivityResponse>());
+            }
+
+            var result = activities.Select(x => new ActivityResponse
+            {
+                Id = x.Id,
+                Duration = x.Duration,
+                TotalCapacity = x.TotalCapacity,
+                LeaderId = x.Leader?.Id ?? 0,
+                StartTime = x.StartTime,
+                EndTime = x.EndTime,
+                Name = x.Name,
+                ShortDescription = x.ShortDescription, 
+                LongDescription = x.LongDescription,
+                Cron = x.Cron,
+                LeaderName = x.Leader?.Name ?? "Unknown"
```

Comment:
I'm not sure about this "Unknown" here, can be null value as well. What would be better?

Reply:
I prefer null. Frontend need to handle it. We just provide data